src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -351,21 +355,18 @@ void add_task_to_actor_queue(LocalSchedulerState *state,
   CHECK(task_counter >= entry->task_counter);
 
   /* Create a new task queue entry. */
-  task_queue_entry *elt = (task_queue_entry *) malloc(sizeof(task_queue_entry));
-  elt->spec = (TaskSpec *) malloc(task_spec_size);
-  memcpy(elt->spec, spec, task_spec_size);
-  elt->task_spec_size = task_spec_size;
+  TaskQueueEntry elt = TaskQueueEntry_init(spec, task_spec_size);
```

Comment:
C++ way would be to just define the constructor `TaskQueueEntry::TaskQueueEntry` appropriately; no `init` methods anywhere.